src/main/scala/mesosphere/marathon/core/instance/GoalChangeReason.scala

```diff
@@ -0,0 +1,30 @@
+package mesosphere.marathon
+package core.instance
+
+/**
+  * Enumeration for the reason why an instance's goal has been adjusted.
+  */
+sealed trait GoalChangeReason extends Product with Serializable
+
+object GoalChangeReason {
```

Comment:
This is super neat; I was wondering what pre-empted the introduction of this feature earlier and it seems like it's introduced because KillReason is gone?

Reply:
Yes, exactly.